### PR TITLE
Use up-to-date fixtures in tests

### LINKED ISSUE #757: Use up-to-date fixtures in tests
Some of our automated tests use project fixtures. A lot of those are outdated. For example, some of those tests use project fixtures that have outdated `cordova-android` versions with bundled dependencies installed. Running our tests against these outdated fixtures does not reflect real-world usage.

Instead, a project as created by the currently running version of Cordova should be used.

CC @erisu

### PR DESCRIPTION
**Pull request by raphinesse**
### Motivation and Context

- Ensures we test `cordova-lib` against our latest platform releases/project structures
- No more outdated versions of our platforms committed to source control
  - Tracked files down from 1725 to 387
  - ~170k LOC less in source control

Resolves #757 

### Description

This introduces a central helper to get test fixtures from. The fixtures cannot be directly accessed by tests but must instead be copied to a target directory specified by the test. The fixtures (mostly projects of different nature) are created using our own tools and up-to-date platforms upon first requesting them. Later requests for the same fixture re-use the formerly created instance to reduce test duration.

All outdated and now unused "static" fixtures have been deleted.

`cordova-test-platform` is required as a dev dependency but does not seem to be released at all. We could either do that, or at least tag a new version in the repository and reference that. I'm open to suggestions.

### CODE REVIEW

**Comment by raphinesse** [comment]
Hmmm, seems that Travis won't run for draft PRs :thinking:

**Comment by raphinesse** [comment]
If anyone with a Windows Machine could have a look at that Node 6 error that occasionally pops up, that would be great.

**Comment by erisu** [comment]
@raphinesse the node issue you are seeing is a known issue with the version of `fs-extra` we have set for this repo.

https://github.com/apache/cordova/issues/121

In the draft branch that I been preparing for the next major bumps this dependency to the current released version that was promising to resolve this issue.

Were you planning to release this PR for the next major or in a minor?

If next major, I can start creating PRs from the commits from my draft branch and merge them in. This would resolve the issue but as well we stop testing Node 6 and 8 in the next major.

File: spec/cordova/fixtures/cordova-android-5.2.2-version.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env node
```
Comment (erisu) [inline spec/cordova/fixtures/cordova-android-5.2.2-version.js]: Add to Header the Apache Software Foundation License

**Comment by erisu** [comment]
Also the CI tests were failing for missing `shelljs` module. I think after that one test is fixed this is good to be merged.

**Comment by raphinesse** [comment]
@erisu thanks for the review. I'll try to find out what's going on with the `shelljs` dependency errors.

The changes from this commit should be able to go in any upcoming release.

**Comment by raphinesse** [comment]
OK. So I'm starting to pull out my hair over this one. [That one error](https://travis-ci.org/apache/cordova-lib/jobs/595781930#L761) is very mysterious. It happens consistently in Travis runs of _this PR_.  I cannot reproduce it locally, neither could @erisu. When I pushed the exact same commit to another branch in my fork, the error vanished and I could not even once reproduce it there.

Right now, I'm at a loss here.

**Comment by raphinesse** [comment]
I don't know what caused the error but I've been able to avoid it. by using a different fixture for the test in question. There's still the Node 6 failure though.

**Comment by raphinesse** [comment]
> If next major, I can start creating PRs from the commits from my draft branch and merge them in. This would resolve the issue but as well we stop testing Node 6 and 8 in the next major.

Yes please! Let's target next major. This will make a few things easier moving forward.

**Comment by raphinesse** [comment]
OK, nice. So now we are left with only one `fs-extra` related error for Node 6 on Windows. That's good enough for me to merge.

**Pull request merged**